### PR TITLE
Blog post: Insights from the OpenTelemetry Docs Usability Survey

### LINKED ISSUE #5793: New Blog Post: Insights from the OpenTelemetry Docs Usability Survey
This blog post will summarize and visualize [the results](https://docs.google.com/spreadsheets/d/1kpJQYiEGtpZorICbl-QfYL3mKfeoRLiUywvKcV8fcNA/edit?resourcekey=&gid=439437959#gid=439437959) from the OTel docs usability survey conducted by the End-User SIG. 

Related SIGs: Communications, End-User

Sponsor: @svrnm @chalin

### PR DESCRIPTION
**Pull request by tiffany76**
This PR creates a blog post that analyzes the results of the recent docs usability survey conducted by the End User SIG. 

Fixes #5793.

Relates to https://github.com/open-telemetry/sig-end-user/issues/69.

**Preview**: https://deploy-preview-5797--opentelemetry.netlify.app/blog/2024/otel-docs-survey/

### CODE REVIEW

File: content/en/blog/2024/otel-docs-survey/index.md [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+linkTitle: OpenTelemetry Docs Survey
+date: 2024-12-13
+author: '[Tiffany Hrabusa](https://github.com/tiffany76) (Grafana Labs)'
+issue: https://github.com/open-telemetry/opentelemetry.io/issues/5793
+sig: Communications, End-User
+# prettier-ignore
+cSpell:ignore: Hrabusa
+---
+
+The OpenTelemetry End-User SIG recently surveyed the community to find out how
```
Comment (svrnm) [inline content/en/blog/2024/otel-docs-survey/index.md:12]:
```suggestion
[The OpenTelemetry End-User SIG](/community/end-user/) recently surveyed the community to find out how
```

File: content/en/blog/2024/otel-docs-survey/index.md [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+user-friendly OpenTelemetry’s documentation is. In an earlier survey, two-thirds
```
Comment (svrnm) [inline content/en/blog/2024/otel-docs-survey/index.md:13]:
```suggestion
user-friendly [OpenTelemetry’s documentation](/docs/) is. In an earlier survey, two-thirds
```

File: content/en/blog/2024/otel-docs-survey/index.md [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+
+## Key takeaways
+
+- Folks want more visual aids like diagrams and screenshots.
+- Of the three types of documentation we asked about (component concepts,
+  installation instructions, and troubleshooting), the troubleshooting docs need
+  the most work.
+- When asked what information they’d most like to see added to OTel’s docs, the
+  top answers were more examples and more coverage, both in depth and breadth.
+- The Collector docs were the most consulted docs set based on number of
```
Comment (svrnm) [inline content/en/blog/2024/otel-docs-survey/index.md:31]:
```suggestion
- The [Collector docs](/docs/collector/) were the most consulted docs set based on number of
```

Do you want to add a sentence that this is also what we see in our google analytics numbers? https://lookerstudio.google.com/s/tSTKxK1ECeU

File: content/en/blog/2024/otel-docs-survey/index.md [hunk 4]
```diff
@@ -0,0 +1,145 @@
...
+  ratings.
+- The Java documentation was the overall highest rated docs, after normalization
```
Comment (svrnm) [inline content/en/blog/2024/otel-docs-survey/index.md:33]:
```suggestion
- The [Java documentation](/docs/languages/java/) was the overall highest rated docs, after normalization
```

looks like the work put into it is honored, cc @zeitlinger @jeanbisutti @jack-berg @trask @jaydeluca

File: content/en/blog/2024/otel-docs-survey/index.md [hunk 5]
```diff
@@ -0,0 +1,145 @@
...
+  and weighting, while the Swift docs received the overall lowest rating
```
Comment (svrnm) [inline content/en/blog/2024/otel-docs-survey/index.md:34]:
```suggestion
  and weighting, while the [Swift](/docs/languages/swift/) docs received the overall lowest rating
```

File: content/en/blog/2024/otel-docs-survey/index.md [hunk 6]
```diff
@@ -0,0 +1,145 @@
...
+- Among the six most popular docs sets, the JavaScript docs received the lowest
```
Comment (svrnm) [inline content/en/blog/2024/otel-docs-survey/index.md:35]:
```suggestion
- Among the six most popular docs sets, the [JavaScript](/docs/languages/js/) docs received the lowest
```

**Review by svrnm — CHANGES_REQUESTED** [review]
This is a really great summary, thanks @tiffany76 !

@open-telemetry/governance-committee @open-telemetry/technical-committee
@open-telemetry/sig-end-user-approvers @open-telemetry/sig-developer-experience-approvers @open-telemetry/sig-contributor-experience-approvers you all might be interested in the preview of these results as well!

**Review by yurishkuro — APPROVED** [review]

File: content/en/blog/2024/otel-docs-survey/index.md [hunk 7]
```diff
@@ -0,0 +1,145 @@
+---
+title: Insights from the OpenTelemetry Docs Usability Survey
+linkTitle: OpenTelemetry Docs Survey
```
Comment (chalin) [inline content/en/blog/2024/otel-docs-survey/index.md:3]:
```suggestion
linkTitle: OTel-docs Survey
```

File: content/en/blog/2024/otel-docs-survey/index.md [hunk 8]
```diff
@@ -0,0 +1,145 @@
...
+date: 2024-12-13
+author: '[Tiffany Hrabusa](https://github.com/tiffany76) (Grafana Labs)'
+issue: https://github.com/open-telemetry/opentelemetry.io/issues/5793
+sig: Communications, End-User
+# prettier-ignore
```
Comment (chalin) [inline content/en/blog/2024/otel-docs-survey/index.md:8]:
```suggestion
```

File: content/en/blog/2024/otel-docs-survey/index.md (hunk 3 above)
Comment (chalin) [reply content/en/blog/2024/otel-docs-survey/index.md:31]: And maybe cross-reference to the corresponding section of your year-in-review post?

- https://deploy-preview-5775--opentelemetry.netlify.app/blog/2024/year-in-review/#which-pages-were-the-most-popular
- Which would correspond to the path `../year-in-review/#which-pages-were-the-most-popular`

File: content/en/blog/2024/otel-docs-survey/index.md (hunk 4 above)
Comment (chalin) [reply content/en/blog/2024/otel-docs-survey/index.md:33]: Very nice! Might again be worth cross-referencing our two posts.

**Review by chalin — COMMENTED** [review]
Initial comments. After @svrnm's suggestions are addressed I'll make another pass.

The text in the images seems faded. Can that easily be corrected? If not, they can stay as is. Thanks.

**Comment by tiffany76** [comment]
> The text in the images seems faded. Can that easily be corrected? If not, they can stay as is. Thanks.

@chalin I made the image text a little darker. Hopefully that addresses your concern. Thanks!

File: content/en/blog/2024/otel-docs-survey/index.md [hunk 9]
```diff
@@ -0,0 +1,152 @@
...
+  Communications'
+  [year-end review](../year-in-review/#which-pages-were-the-most-popular).
+- After normalization and weighting, the
+  [Java documentation](/docs/languages/java/) was the overall highest rated docs
+  set, suggesting that the efforts made to
+  [improve its information architecture](../year-in-review/#ia-improvements)
+  have paid off. Meanwhile, the [Swift](/docs/languages/swift/) docs received
+  the overall lowest rating.
+- Among the six most popular docs sets, the [JavaScript](/docs/languages/js/)
+  docs received the lowest rating.
```
Comment (chalin) [inline content/en/blog/2024/otel-docs-survey/index.md:43]: How about the following:

```suggestion
- Respondents expressed a desire for **more visual aids**, such as diagrams and
  screenshots.
- Of the three types of documentation we asked about (component concepts,
  installation instructions, and troubleshooting), the **troubleshooting docs**
  were identified as needing the most improvement.
- When asked about the information they’d most like to see added to OTel's docs,
  the top responses were **more examples** and **expanded coverage**, both in
  depth and breadth.
- The [Collector](/docs/collector/) docs emerged as the most frequently
  consulted resource, a finding that aligns with the page view analysis in the
  SIG Communications'
  [year-end review](../year-in-review/#which-pages-were-the-most-popular).
- After normalization and weighting, the [Java](/docs/languages/java/)
  documentation received the **highest overall rating**, reflecting the positive
  impact of recent improvements to its
  [organization](../year-in-review/#ia-improvements). Conversely, the
  [Swift](/docs/languages/swift/) docs received the lowest overall rating.
- Among the six most popular documentation sets, the
  [JavaScript](/docs/languages/js/) docs received the **lowest rating**.
```

**Review by chalin — COMMENTED** [review]
Looking good! See inline for suggested revisions. Thanks for putting this together!

File: content/en/blog/2024/otel-docs-survey/index.md [hunk 10]
```diff
@@ -0,0 +1,152 @@
...
+
+#### Q: What source do you primarily rely on when you're looking for information about OpenTelemetry?
+
+- Overall, the majority of respondents (52%) rely on the opentelemetry.io
+  documentation.
+- Respondents early in their observability practice (beginner and intermediate)
+  are more likely to use the opentelemetry.io documentation.
+- Expert observability practitioners prefer the code repository documentation.
+
+![Percentage of website users by observability knowledge level](website-usage.png)
```
Comment (chalin) [inline content/en/blog/2024/otel-docs-survey/index.md:62]: How about replacing this image with markdown link this (it makes for a better viewing experience in dark mode):

```suggestion
> Respondents who use [opentelemetry.io] as their
> primary information source<br>_By level of observability knowledge_
>
> | **Beginners** | **Intermediates** | **Experts** |
> | :-----------: | :---------------: | :---------: |
> |     100%      |        62%        |     44%     |
```

Here's what it looks like:

> image

Comment (chalin) [reply content/en/blog/2024/otel-docs-survey/index.md:62]: If you agree, then pls drop the corresponding image.

File: content/en/blog/2024/otel-docs-survey/index.md [hunk 11]
```diff
@@ -0,0 +1,152 @@
...
+
+- The Java documentation has the highest overall rating.
+- The Swift documentation has the lowest overall rating.
+
+![Table showing the number of ratings (poor, okay, great) by OpenTelemetry documentation set](ratings-table.png)
+
+If we combine these insights, we can see that the documentation used by the most
+people that needs the most work is the JavaScript documentation.
+
+![Subset of the previous table showing the ratings for the six most popular documentation sets](javascript-table.png)
```
Comment (chalin) [inline content/en/blog/2024/otel-docs-survey/index.md:136]: Again, how about using a markdown table here?

```suggestion
> **Where should we focus our improvement efforts?**<br> _JavaScript is one of
> the six most-used docs sets, but its rating is the lowest._
>
> | Language or component | Poor | Okay | Great | Total responses | Normalized & weighted |
> | :-------------------- | :--: | :--: | :---: | :-------------: | :-------------------: |
> | GO                    |  6   |  12  |   9   |       27        |        7.1111         |
> | Java                  |  3   |  16  |   8   |       27        |        7.3333         |
> | JavaScript            |  3   |  19  |   2   |       24        |        6.0000         |
> | Python                |  6   |  17  |   8   |       31        |        6.7742         |
> | Collector             |  9   |  17  |  11   |       37        |        6.8108         |
> | Kubernetes            |  6   |  20  |   6   |       32        |        6.3750         |
```

If you agree, drop the corresponding image. Here's what it looks like:

image

File: content/en/blog/2024/otel-docs-survey/index.md [hunk 12]
```diff
@@ -0,0 +1,152 @@
...
+
+![Horizontal bar graph showing the number of ratings per OTel component](top-six-bar.png)
+
+Here are the tabulated ratings for all languages and components. When the
+results are normalized and weighted, we can see additional insights:
+
+- The Java documentation has the highest overall rating.
+- The Swift documentation has the lowest overall rating.
+
+![Table showing the number of ratings (poor, okay, great) by OpenTelemetry documentation set](ratings-table.png)
```
Comment (chalin) [inline content/en/blog/2024/otel-docs-survey/index.md:131]: How about:

```suggestion
> **How would you rate your experience using the current OTel documentation for
> the following languages and components?**
>
> | Language or component | Poor | Okay | Great | Total responses | Normalized & weighted |
> | :-------------------- | :--: | :--: | :---: | :-------------: | :-------------------: |
> | .NET                  |  4   |  8   |   2   |       14        |        5.7143         |
> | C++                   |  0   |  7   |   0   |        7        |        6.0000         |
> | Erlang                |  1   |  6   |   0   |        7        |        5.4286         |
> | GO                    |  6   |  12  |   9   |       27        |        7.1111         |
> | Java                  |  3   |  16  |   8   |       27        |      **7.3333**       |
> | JavaScript            |  3   |  19  |   2   |       24        |        6.0000         |
> | PHP                   |  1   |  4   |   2   |        7        |        7.1429         |
> | Python                |  6   |  17  |   8   |       31        |        6.7742         |
> | Ruby                  |  1   |  5   |   1   |        7        |        6.2857         |
> | Rust                  |  4   |  4   |   2   |       10        |        5.6000         |
> | Swift                 |  3   |  3   |   0   |        6        |      **4.0000**       |
> | Collector             |  9   |  17  |  11   |       37        |        6.8108         |
> | Kubernetes            |  6   |  20  |   6   |       32        |        6.3750         |
> | FaaS                  |  5   |  7   |   0   |       12        |        4.3333         |
> | **Total**             |  52  | 145  |  51   |                 |                       |
```

I think that we should sort by the value of the final column, like this:

```suggestion
> **How would you rate your experience using the current OTel documentation for the following languages and components?**
>
> | Language or component | Poor | Okay | Great | Total responses | Normalized & weighted |
> |:-----------------------|:----:|:----:|:-----:|:---------------:|:----------------------:|
> | Java                  | 3    | 16   | 8     | 27              | 7.3333                |
> | PHP                   | 1    | 4    | 2     | 7               | 7.1429                |
> | GO                    | 6    | 12   | 9     | 27              | 7.1111                |
> | Collector             | 9    | 17   | 11    | 37              | 6.8108                |
> | Python                | 6    | 17   | 8     | 31              | 6.7742                |
> | Kubernetes            | 6    | 20   | 6     | 32              | 6.3750                |
> | C++                   | 0    | 7    | 0     | 7               | 6.0000                |
> | JavaScript            | 3    | 19   | 2     | 24              | 6.0000                |
> | Ruby                  | 1    | 5    | 1     | 7               | 6.2857                |
> | Rust                  | 4    | 4    | 2     | 10              | 5.6000                |
> | .NET                  | 4    | 8    | 2     | 14              | 5.7143                |
> | Erlang                | 1    | 6    | 0     | 7               | 5.4286                |
> | FaaS                  | 5    | 7    | 0     | 12              | 4.3333                |
> | Swift                 | 3    | 3    | 0     | 6               | 4.0000                |
> | **Total**             | 52   | 145  | 51    |                 |                      |
```

**Comment by chalin** [comment]
All checks are passing now that this PR has been rebased (FYI).

File: content/en/blog/2024/otel-docs-survey/index.md (hunk 9 above)
Comment (tiffany76) [reply content/en/blog/2024/otel-docs-survey/index.md:43]: If the highest ratings are takeaways, I think the lowest ones are too. I think we should keep them, but I'm also fine removing them if you think we should.

File: content/en/blog/2024/otel-docs-survey/index.md (hunk 10 above)
Comment (tiffany76) [reply content/en/blog/2024/otel-docs-survey/index.md:62]: Thanks. I didn't think about dark mode. I'll make the change.

**Review by chalin — APPROVED** [review]
Looking great! Thanks for the updates. 🚀

**Review by svrnm — APPROVED** [review]

File: content/en/blog/2024/otel-docs-survey/index.md [hunk 13]
```diff
@@ -0,0 +1,191 @@
+---
+title: Insights from the OpenTelemetry Docs Usability Survey
+linkTitle: OTel-docs Survey
+date: 2024-12-13
```
Comment (chalin) [inline content/en/blog/2024/otel-docs-survey/index.md:4]:
```suggestion
date: 2024-12-18
```

**Pull request merged**